packages/ui-components/src/components/toggle-tip/toggle-tip.tsx

```diff
@@ -0,0 +1,87 @@
+import { autoPlacement, computePosition, ComputePositionConfig, Middleware, offset, shift } from '@floating-ui/dom';
+import { Component, Host, h, Prop, Element, State } from '@stencil/core';
+import { isEmpty } from 'lodash-es';
+import { EIconName } from '../icon/icon.types';
+import { DEFAULT_AUTOPLACEMENT_CONFIG, DEFAULT_OFFSET, DEFAULT_SHIFT_CONFIG, DEFAULT_TOGGLETIP_CONFIG } from './toggle-tip.config';
+import { EToggleTipPosition } from './toggle-tip.types';
+
+/**
+ * @part toggle-tip-container - The toggle tip container.
+ * @part toggle-tip-slot-content - The toggle tip slot content.
+ */
+@Component({
+	tag: 'kv-toggle-tip',
+	styleUrl: 'toggle-tip.scss',
+	shadow: true
+})
+export class KvToggleTip {
```

Comment:
Isn't possible to extend the `KvToggleTip` by `KvTooltip`?
Or create a generic class that have pratically all the code that are shared between the compoents and extend both class that component @GoncaloCamaz ?

Reply:
@miguelpinto98 i added now some code in order to check for clicks outside of the toggle tip. Therefore and since the behaviours of this two components are distinct regarding opening and closing, i believe its better to separate both but please feel free to suggest otherwise. 
I will try to extend the tooltip props for the toggle tip.